samples/RoutingSample.Web/HelloExtension/HelloAppBuilderExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Options;
+using RoutingSample.Web.HelloExtension;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class HelloAppBuilderExtensions
+    {
+        public static IApplicationBuilder UseHello(this IApplicationBuilder app, string greeter)
```

Comment:
Note that there's no real reason for this level of indirection - unless you're doing it for sampley reasons 😆

Reply:
I figured you did it this way because health check already had middleware, but I left it like this to functional test this approach worked.

Reply:
alright dope